src/components/Orderbook.tsx

```diff
@@ -1,196 +1,414 @@
+          if (abortCtrl.signal.aborted) return
+
+          setOrders(orders)
+          setAlert(null)
+        })
+        .catch((e) => {
+          if (abortCtrl.signal.aborted) return
+          const message = t('orderbook.error_loading_orderbook_failed', {
+            reason: e.message || 'Unknown reason',
+          })
+          setAlert({ variant: 'danger', message })
+        })
+    },
+    [t]
+  )
+
   useEffect(() => {
-    setIsLoading(true)
+    if (!show) return
 
     const abortCtrl = new AbortController()
 
-    ObwatchApi.fetchOrderbook({ signal: abortCtrl.signal })
-      .then((orders) => {
-        if (abortCtrl.signal.aborted) return
-        setOrders(orders)
-      })
-      .catch((e) => {
-        if (abortCtrl.signal.aborted) return
-        const message = t('orderbook.error_loading_orderbook_failed', {
-          reason: e.message || 'Unknown reason',
-        })
-        setAlert({ variant: 'danger', message })
-      })
-      .finally(() => {
-        if (abortCtrl.signal.aborted) return
-        setIsLoading(false)
-        setIsInitialized(true)
-      })
+    setIsLoading(true)
+    refresh(abortCtrl).finally(() => {
+      if (abortCtrl.signal.aborted) return
+      setIsLoading(false)
+      setIsInitialized(true)
+    })
 
     return () => {
       abortCtrl.abort()
     }
-  }, [t])
+  }, [show, refresh])
 
   return (
-    <div>
-      {!isInitialized && isLoading ? (
-        Array(5)
-          .fill('')
-          .map((_, index) => {
-            return (
-              <rb.Placeholder key={index} as="div" animation="wave">
-                <rb.Placeholder xs={12} className={styles['orderbook-line-placeholder']} />
-              </rb.Placeholder>
-            )
-          })
-      ) : (
-        <>
-          {alert && <rb.Alert variant={alert.variant}>{alert.message}</rb.Alert>}
-          {orders && (
-            <rb.Row>
-              <rb.Col className="mb-3">
-                <OrderbookTable orders={orders} />
-              </rb.Col>
-            </rb.Row>
-          )}
-        </>
-      )}
-    </div>
-  )
-}
-
-export function OrderbookOverlay({ show, onHide }: rb.OffcanvasProps) {
-  const { t } = useTranslation()
-
-  return (
-    <rb.Offcanvas className={styles['orderbook-overlay']} show={show} onHide={onHide} placement="bottom">
-      <rb.Offcanvas.Header closeButton>
-        <rb.Offcanvas.Title>{t('orderbook.title')}</rb.Offcanvas.Title>
+    <rb.Offcanvas
+      className={`offcanvas-fullscreen ${styles.overlayContainer}`}
+      show={show}
+      onHide={onHide}
+      placement="bottom"
+    >
+      <rb.Offcanvas.Header>
+        <rb.Container>
+          <div className="w-100 d-flex">
+            <div className="d-flex align-items-center flex-1">
+              <rb.Offcanvas.Title>{t('orderbook.title')}</rb.Offcanvas.Title>
+            </div>
+            <div>
+              <rb.Button variant="link" className="unstyled pe-0 ms-auto me-auto me-md-0" onClick={onHide}>
```

Comment:
```suggestion
              <rb.Button variant="link" className="unstyled pe-0" onClick={onHide}>
```

Same as for the earn report above.